--- a/Makefile
+++ b/Makefile
@@ -1,25 +1,20 @@
-CC?=gcc
-INCLUDES?=-I/usr/local/include
-LIBDIRS?=-L/usr/local/lib
-LIB?=-lz -lcrypto
+CC := gcc
+LIBS := -lssl -lz -lcrypto
 
 OS=$(shell uname)
 ifneq (,$(findstring BSD,$(OS)))
 	# hopefully FreeBSD, OpenBSD, NetBSD
-	CFLAGS+=-DBSD -std=c99 -D_FILE_OFFSET_BITS=64 -O2 -Wall -g $(INCLUDES)
-	PREFIX?=/usr/local
-	TINCDIR?=/usr/local/etc/tinc
+	PREFIX=/usr/local
+	TINCDIR=/usr/local/etc/tinc
 else
 ifneq (,$(findstring Darwin,$(OS)))
 	# Apple OSX / Darwin
-	CFLAGS+=-DBSD -std=c99 -D_FILE_OFFSET_BITS=64 -O2 -Wall -g $(INCLUDES)
-	PREFIX?=/usr/local
-	TINCDIR?=/usr/local/etc/tinc
+	PREFIX=/usr/local
+	TINCDIR=/usr/local/etc/tinc
 else
 	# Linux by default
-	CFLAGS+=-std=c99 -D_POSIX_C_SOURCE=2 -D_BSD_SOURCE -D_FILE_OFFSET_BITS=64 -O2 -Wall -g $(INCLUDES)
-	PREFIX?=/usr
-	TINCDIR?=/etc/tinc
+	PREFIX=/usr
+	TINCDIR := /etc/tinc
 endif
 endif
 ifneq (,$(findstring mingw,$(CC)))
@@ -27,10 +22,9 @@ ifneq (,$(findstring mingw,$(CC)))
 	LIB+=-lws2_32 -lgdi32
 endif
 
-LEX=flex
-YACC?=yacc
-
-CFLAGS += -DPREFIX="\"$(PREFIX)\"" -DTINCDIR="\"$(TINCDIR)\""
+LEX := flex
+YACC := yacc
+CFLAGS += '-DTINCDIR="\"$(TINCDIR)\""'
 
 STRINGSRC=string/string_clear.c string/string_concatb.c string/string_concat_sprintf.c string/string_putc.c string/string_putint.c string/string_concat.c string/string_free.c string/string_get.c string/string_init.c string/string_equals.c string/string_move.c string/string_initfromstringz.c string/string_lazyinit.c string/string_read.c string/string_hexdump.c
 HTTPLIBSRC=httplib/http_get.c httplib/http_parseurl.c
@@ -40,19 +34,19 @@ STRINGOBJ=$(patsubst %.c,%.o,$(STRINGSRC
 HTTPLIBOBJ=$(patsubst %.c,%.o,$(HTTPLIBSRC))
 OBJ=$(patsubst %.c,%.o,$(SRC))
 
-NAME?=chaosvpn
+NAME=chaosvpn
 GITDEBVERSION=$(shell debian/scripts/calcdebversion )
 
 all: $(NAME)
 
 $(NAME): main.o $(OBJ) $(HEADERS)
-	$(CC) $(LDFLAGS) -o $@ main.o $(OBJ) $(LIB) $(LIBDIRS)
+	$(CC) $(COPT) -o $@ main.o $(OBJ) $(LIBS) $(LIB)
 
 test_addrmask: test_addrmask.o $(OBJ) $(HEADERS)
 	$(CC) $(LDFLAGS) -o $@ test_addrmask.o $(OBJ) $(LIB) $(LIBDIRS)
 
 %.o: %.c $(HEADERS)
-	$(CC) $(CFLAGS) -o $(patsubst %.c,%.o,$<) -c $<
+	$(CC) $(CFLAGS) $(COPT) -o $(patsubst %.c,%.o,$<) -c $<
 
 lex.yy.o: lex.yy.c y.tab.h
 
