--- a/Makefile.in
+++ b/Makefile.in
@@ -44,12 +44,12 @@ PACKAGE_NAME = @PACKAGE_NAME@
 PACKAGE_VERSION = @PACKAGE_VERSION@
 
 prefix = @prefix@
-exec_prefix = @exec_prefix@
+exec_prefix += @exec_prefix@
 
 bindir = $(exec_prefix)/bin
 
 # Where to put the manual pages.
-mandir = @mandir@
+mandir = $(exec_prefix)/@mandir@
 man1dir = $(mandir)/man1
 # Extension (including `.') for the manual page filenames.
 man1ext = .1
