--- a/mDNSPosix/Makefile
+++ b/mDNSPosix/Makefile
@@ -362,12 +362,16 @@ STRIP = strip
 endif
 else
 
-ifeq ($(os),linux)
+# Any variant containing linux:
+ifeq ($(findstring linux,$(os)),linux)
 CFLAGS_OS = -DNOT_HAVE_SA_LEN -DUSES_NETLINK -DHAVE_LINUX -DTARGET_OS_LINUX
 FLEXFLAGS_OS = -l
 JAVACFLAGS_OS += -I$(JDK)/include/linux
+# uClibc does not provide NSS, do not compile nss_mdns:
+ifneq ($(os),linux-uclibc)
 OPTIONALTARG = nss_mdns
 OPTINSTALL   = InstalledNSS
+endif
 else
 
 ifeq ($(os),netbsd)
