--- a/mDNSShared/dnsextd_parser.y
+++ b/mDNSShared/dnsextd_parser.y
@@ -56,7 +56,7 @@ Revision 1.1  2006/07/06 00:09:05  chesh
 #include "DebugServices.h"
 #include "dnsextd.h"
 
-void yyerror( const char* error );
+void yyerror( void *context, const char* error );
 int  yylex(void);
 
 
@@ -114,8 +114,6 @@ static ZoneSpec			*	g_zones;
 static ZoneSpec				g_zoneSpec;
 static const char		*	g_filename;
 
-#define YYPARSE_PARAM  context
-
 void
 SetupOptions
 	(
@@ -125,6 +123,8 @@ SetupOptions
 
 %}
 
+%parse-param { void *context }
+
 %union
 {
 	int			number;
@@ -411,7 +411,7 @@ int yywrap(void);
 
 extern int yylineno;
 
-void yyerror( const char *str )
+void yyerror( void *context, const char *str )
 {
         fprintf( stderr,"%s:%d: error: %s\n", g_filename, yylineno, str );
 }
