--- a/configure.ng
+++ b/configure.ng
@@ -93,35 +93,6 @@ AC_DEFUN([GCC_STACK_PROTECT_CC],[
   fi
 ])
 
-AC_DEFUN([WORKING_GETADDRINFO],[
-	AC_CHECK_FUNCS([getaddrinfo],[
-		AC_MSG_CHECKING([whether getaddrinfo() works])
-		AC_TRY_RUN([
-#include <stdio.h>
-#include <sys/types.h>
-#include <sys/socket.h>
-#include <netdb.h>
-int
-main(int argc, char **argv)
-{
-	struct addrinfo hints, *ai;
-	memset(&hints, 0, sizeof(hints));
-	hints.ai_flags = AI_PASSIVE;
-	hints.ai_socktype = SOCK_STREAM;
-	hints.ai_family = PF_UNSPEC;
-	if(getaddrinfo(NULL, "0", &hints, &ai) != 0)
-		return 1;
-	return 0;
-}
-		],[
-		AC_DEFINE([HAVE_WORKING_GETADDRINFO], 1, [getaddrinfo(0)])
-		AC_MSG_RESULT(yes)
-		],[
-		AC_MSG_RESULT(no)
-		])
-	])
-])
-
 if test "$GCC" = "yes"; then
 	# We are using the GNU C compiler. Good!
 	CFLAGS="$CFLAGS -pipe -W -Wall -Wpointer-arith -Wstrict-prototypes"
@@ -217,11 +188,10 @@ AC_CHECK_FUNCS([ \
 
 # Optional functions
 AC_CHECK_FUNCS_ONCE([
-	arc4random arc4random_stir gai_strerror getnameinfo inet_aton \
+	arc4random arc4random_stir gai_strerror getaddrinfo getnameinfo inet_aton \
 	sigaction sigprocmask snprintf vsnprintf strdup strndup strlcpy strlcat \
 	strtok_r unsetenv waitpid])
 
-WORKING_GETADDRINFO
 
 # -- Configuration options --
 
--- a/src/ngircd/resolve.c
+++ b/src/ngircd/resolve.c
@@ -242,7 +242,7 @@ ForwardLookup(const char *hostname, arra
 {
 	ng_ipaddr_t addr;
 
-#ifdef HAVE_WORKING_GETADDRINFO
+#ifdef HAVE_GETADDRINFO
 	int res;
 	struct addrinfo *a, *ai_results;
 	static struct addrinfo hints;
--- a/src/ipaddr/ng_ipaddr.c
+++ b/src/ipaddr/ng_ipaddr.c
@@ -23,7 +23,7 @@
 GLOBAL bool
 ng_ipaddr_init(ng_ipaddr_t *addr, const char *ip_str, UINT16 port)
 {
-#ifdef HAVE_WORKING_GETADDRINFO
+#ifdef HAVE_GETADDRINFO
 	int ret;
 	char portstr[64];
 	struct addrinfo *res0;
